### DISCUSSION TITLE
Exclude specific version of dependency

### DISCUSSION THREAD

**Discussion by caclark** [mesonbuild/meson#11884] [Q&A]
How can I exclude a specific version of a dependency, for example an equivalent of the following:
`req_version = ['>=0.11', '!=0.28.0']`

Thanks..

**Comment by xclaesse** [comment]
Just like you said: `dependency('glib-2.0', version: ['>=2.0', '!= 2.76.1'])`

Result:
```
Dependency glib-2.0 found: NO found 2.76.1 but need: '!= 2.76.1' ; matched: '>=2.0'

```

**Comment by caclark** [comment]
Ah, I see that the problem occurs when I exclude the only version available on the system - in this case exiv2 0.27.6 is what I have installed.
When I use:
`req_version = ['>=0.11', '!=0.27.6']`

I get:
`Dependency exiv2 found: NO found 0.27.6 but need: '!=0.27.6' ; matched: '>=0.11'`
`Found CMake: /usr/bin/cmake (3.25.1)`
`Run-time dependency exiv2 found: NO (tried pkgconfig)`
`meson.build:287:8: ERROR: The `+` operator of str does not accept objects of type list (['>=0.11', '!=0.27.6'])`

**Reply by xclaesse** [reply]
What do you have at line 287?

**Comment by caclark** [comment]
Thanks. I had a severe brain fade moment and mis-read the line number.
Sorry to have troubled you.
